### PR TITLE
add package commands

### PR DESCRIPTION
**Pull request by lamebrain**
* refactored dependency injections for services and logger
* added appsettings.json
* package commands
  * create
  * upload
  * list
  * publish

### CODE REVIEW

File: Meadow.CLI/Properties/launchSettings.json [hunk 1]
```diff
@@ -30,7 +30,7 @@
     },
     "FlashEsp": {
       "commandName": "Project",
-      "commandLineArgs": "flash esp"
+      "commandLineArgs": "flash esp -v 0.9.6.1"
```
Comment (CartBlanche) [inline Meadow.CLI/Properties/launchSettings.json:33]: Do we want the version number hard coded here?

**Review by patridge — APPROVED** [review]
Not sure I know enough of the project to properly review this, but it looks good at the level I understand it.

File: Meadow.CLI/Properties/launchSettings.json (hunk 1 above)
Comment (lamebrain) [reply Meadow.CLI/Properties/launchSettings.json]: nope! thanks

File: Meadow.CLI.Core/Managers/PackageManager.cs [hunk 2]
```diff
@@ -27,13 +27,13 @@ public string CreatePackage(string applicationPath, string osVersion)
...
             {
-                _logger.LogInformation($"\"{applicationPath}\" is not found. Please try again.");
+                throw new ArgumentException("Invalid applicationPath");
             }
 
             var osFilePath = Path.Combine(DownloadManager.FirmwareDownloadsFilePathRoot, osVersion);
             if (!Directory.Exists(osFilePath))
             {
-                _logger.LogInformation($"Version {osVersion} not found. Please download.");
+                throw new ArgumentException("osVersion not found. Please download.");
```
Comment (CartBlanche) [inline Meadow.CLI.Core/Managers/PackageManager.cs:36]: In this thrown exception and the one above, why wouldn't we want the actual OSVerion or Application Path in the message?

Comment (lamebrain) [reply Meadow.CLI.Core/Managers/PackageManager.cs:36]: we would want it :)

**Comment by lamebrain** [comment]
> @lamebrain I've added a couple of questions, but I don't feel I'm familiar enough with the actual cloud requirements and functionality to comment on that.

Thanks! I've been trying to keep in mind that one day this will also be integrated into the IDEs, hence some of the service/manager refactorings. I've been locally testing a lot of the cloud functionality, so I feel ok about that as long as there isn't anything terribly wrong with the structure/design.

**Pull request merged**